unity-renderer/Assets/UnityGLTF/Scripts/DownloadQueueHandler.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using UnityEngine;
+
+namespace UnityGLTF
+{
+    internal class DownloadQueueHandler : IDisposable
+    {
+        private static event Action OnCompetitionReset;
+        private static GLTFComponent nearestComponent;
+        private static float nearestDistance = float.MaxValue;
+        private static int competingCount;
+        private static bool isCompetitionFinished = false;
```

Comment:
Not sold at all with the design choice of making instances of `DownloadQueueHandler` with the queue competition being tracked by static variables. I know the original was worst though, but extracting the code to another class opens up opportunities to do test coverage of this logic. 

I think that even a singleton can be better than this because we can just instance the class to test it.
We can just make a `DownloadQueueHandler` composite as a singleton, in a way that each `GLTFComponent` creates its own `DownloadQueueHandler` like it does now, but reports it to the composite.

This means that we should add coverage to it too, so I'll add this as a request.

Reply:
I have changed the approach for the handler to make it independent from static variables.
I've added the code for master promises that I had in other PR related to the download queue.
Sending forgotten master promise to the floor will not solve the issue cause we don't know where the pending promises are in world position, so for that cases I can only prioritize the download